src/the-final-count-down/modules/ClockDisplayUnit.js

```diff
@@ -0,0 +1,107 @@
+define([
+    "loDash",
+    "modules/CountDownClock"
+], function(_, CountDownClock){
+
+    /**
+     * A class that listens to events from a CountDownClock and updates cerosComponents when required
+     *
+     * @param {CountDownClock} countDownClock
+     * @param {string} unitsToDisplay
+     * @param {number} paddingZeros
+     * @constructor
+     */
+    var ClockDisplayUnit = function(countDownClock, unitsToDisplay, paddingZeros) {
+
+        // Instance of the clock we're listening to
+        this.countDownClock = countDownClock;
+
+        // The value currently being displayed
+        this.currentValue = null;
+
+        // The cerosComponents to update on change
+        this.cerosComponents = [];
+
+        // The Clock units we're displaying
+        this.unitsToDisplay = unitsToDisplay;
+
+        // The number of 0s to add our value with, ie "1" becomes "01" if the value is 2
+        this.paddingZeros = paddingZeros || 2;
```

Comment:
So this should probably be called something like `paddedLengthOfNumber`, because it's not actually the number of zeroes.